src/common/desktop/ApplicationWindow.ts

```diff
@@ -73,6 +73,7 @@ export class ApplicationWindow {
 		private readonly localShortcut: LocalShortcutManager,
 		private readonly themeFacade: DesktopThemeFacade,
 		private readonly remoteBridge: RemoteBridge,
+		private desktopExportLocks: Set<string>,
```

Comment:
napkin pitch: what if we make the "window cleanup" mechanism more generic and inject some listener here instead? and coordination of these listeners can be outside of the poor `ApplicationWindow`.